src/xor_linked_list.rs

```diff
@@ -0,0 +1,1625 @@
+    /// new list consisting of everything after the cursor, with the original
+    /// list retaining everything before.
+    ///
+    /// If the cursor is pointing at the null object then the entire contents
+    /// of the `XorLinkedList` are moved.
+    #[inline]
+    pub fn split_after(&mut self) -> XorLinkedList<A>
+    where
+        A: Clone,
+    {
+        unsafe {
+            let current = match self.current.as_ref() {
+                Some(current) => current,
+                None => {
+                    let adapter = self.list.adapter.clone();
+                    return mem::replace(&mut self.list, XorLinkedList::new(adapter));
+                }
+            };
+            let head = self.link.next;
+            let tail = if let Some(head) = head.as_ref() {
+                head.replace_ptr(self.current, ptr::null());
+                self.link.next = ptr::null();
+                current.set(self.link);
+                self.list.tail
+            } else {
+                // Empty list after current
+                ptr::null()
+            };
+            let list = XorLinkedList {
+                head,
+                tail,
+                adapter: self.list.adapter.clone(),
+            };
+            self.list.tail = current;
+            list
+        }
+    }
+
+    /// Splits the list into two before the current element. This will return a
+    /// new list consisting of everything before the cursor, with the original
+    /// list retaining everything after.
+    ///
+    /// If the cursor is pointing at the null object then the entire contents
+    /// of the `XorLinkedList` are moved.
+    #[inline]
+    pub fn split_before(&mut self) -> XorLinkedList<A>
+    where
+        A: Clone,
+    {
+        unsafe {
+            let current = match self.current.as_ref() {
+                Some(current) => current,
+                None => {
+                    let adapter = self.list.adapter.clone();
+                    return mem::replace(&mut self.list, XorLinkedList::new(adapter));
+                }
+            };
+            let tail = self.link.prev;
+            let head = if let Some(tail) = tail.as_ref() {
+                tail.replace_ptr(self.current, ptr::null());
+                self.link.prev = ptr::null();
+                current.set(self.link);
+                self.list.head
+            } else {
+                // Empty list before current
+                ptr::null()
+            };
+            let list = XorLinkedList {
+                head,
+                tail,
+                adapter: self.list.adapter.clone(),
+            };
+            self.list.head = current;
+            list
+        }
+    }
+}
+
+// =============================================================================
+// XorLinkedList
+// =============================================================================
+
+/// Intrusive xor doubly-linked list which uses less memory than a regular doubly linked list
```

Comment:
Can you add a quick line here explaining the drawback in exchange for the lower memory use?